products/bridge/test/utils.ts

```diff
@@ -306,19 +356,15 @@ export async function deliverResult(
     );
 
   // other validators see the Resumed event and do not attempt to deliver the result again
-  for (const index of signerIndices) {
-    const blockNum = await ethers.provider.getBlockNumber();
-
+  for (const validator of validators) {
     const filter = relayer.filters.Resumed(
       caller,
       undefined,
       undefined,
       undefined,
       nonce
     );
-    const logs = await relayer
-      .connect(validators[index])
-      .queryFilter(filter, blockNum - 100, blockNum);
+    const logs = await relayer.connect(validator).queryFilter(filter);
```

Comment:
As discussed, it's essential to only consider events in finalized blocks, therefore the code before refactoring fetched `ethers.provider.getBlockNumber('finalized')` and queried only until that block number.